spec/src/main/asciidoc/required-apis.asciidoc

```diff
@@ -56,6 +58,15 @@ Use of implementations beyond CDI 2.0 in MicroProfile is allowed, however, not r
  - https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html
  - http://cdi-spec.org/
 
+==== Optional Requirements
+
+MicroProfile 4.x implementations may opt to not implement the following CDI features:
```

Comment:
```suggestion
The following CDI features have been made optional in the MicroProfile 4.x implementations.
```

Reply:
We can't declare portions of CDI as being optional.  The Jakarta EE CDI team can declare parts of their spec as optional, but MicroProfile can't do that.  We can state the support of CDI in general is optional (per the earlier TCK test requirements).  But, we can't declare portions of the CDI API or Annotations as being optional.

Reply:
Is the scenario that we declare these features optional just in the context of MP possible? Meaning that we "override" this in the TCKs that are being used for MP, not touching the underlying Jakarta EE base?

Reply:
I don't think we can claim CDI as being optional though. CDI is required. EJB, EL part from CDI are optional as we removed that dependency.

Reply:
I think we need to reword: some part of CDI specs are not required in MicroProfile.

Reply:
My understanding from the discussion call was that we were going to list part of the CDI spec that were not required by MP. As stated, it is impossible for a MP only implementation to pass the CDI TCK, because it includes things like EL Support and EJB integration that we explicitly exclude in the specification pom.

In my opinion we are already making these things optional (from the pom), we are just specifying them here. I'm ok to go over this discussion again, but in my opinion the specification pom and the specification wording need to align.

Reply:
By the way, I disagree with the above statement @radcortez . Even though some features such EL, EJB were excluded, runtimes are absolutely fine to add them in their integration. MP should not force MP impls not to integrate EL, EJB, etc. I think we should only assert on these integrations are not required.

Reply:
To clarify, all my considerations are about a MP runtime without additional Jakarta dependencies.

If we say these are optional (or not have to be included), is not forcing implementations to not include these features.